spec/models/tag_spec.rb

```diff
@@ -103,6 +103,18 @@
       expect(categorization).to eq(expected_categorization)
     end
 
+    it "tag with nil classification" do
+      @tag.classification = nil
```

Comment:
Shouldn't this have been `@tag.classification.delete`?

If I do this and issue `@tag.reload`, I get:

```
  1) Tag categorization tag with nil classification
     Failure/Error: expect(@tag.show).to eq(true)

       expected: true
            got: nil

       (compared using ==)
     # ./spec/models/tag_spec.rb:117:in `block (3 levels) in <top (required)>'
```

Note, I'm here because rails 5.0 looks to have been returning a cached classification when `@tag.categorization` calls `@tag.category` in `show`.  Because this was failing, I changed my tests locally to do `@tag.classification.delete` and it fails as shown above in both rails 5.0 and 5.1.

I'm not sure how any of the `@category` expectations are set, such as `@category.name` below:
To get to the category, you need to navigate through using `@tag.classification.parent` (what category does I think).  If the classification is nil or deleted, category should also be nil, which would cause `show` to be nil.

😕

Reply:
In other words, I believe we're relying on `@tag.category` to return the cached parent of the classification's parent.  If you delete the classification, it fails.  If you upgrade to rails 5.1, the classification nil so the category (classification parent) is also nil.

I believe this test is wrong.  I think.